src/loaders/OMETiffLoader.js

```diff
@@ -195,12 +200,17 @@ export default class OMETiffLoader {
         // We need to put the request for parsing the file directory into this array.
         // This allows us to get tiff pages directly based on offset without parsing everything.
         if (!isBioFormats6Pyramid) {
-          if (offsets) {
+          if (offsets.length > 0) {
             tiff.ifdRequests[pyramidIndex] = tiff.parseFileDirectoryAt(
               offsets[pyramidIndex]
             );
           }
         } else {
+          // Pyramids with large z-stacks + large numbers of channels could get slow
+          // so we allow for offsets for the initial images ("parentImage").
+          if (offsets.length > 0) {
+            tiff.ifdRequests[index] = tiff.parseFileDirectoryAt(offsets[index]);
+          }
```

Comment:
Duplicated from the above. Would it be possible to wrap this in a private method with more description name that can be reused?